app/src/main/java/lu/geoportail/map/LuxTileServer.kt

```diff
@@ -399,9 +409,18 @@ class LuxTileServer(
             response.send(json.toString(4))
         }
 
+    private val checkPreflight =
+        HttpServerRequestCallback { request: AsyncHttpServerRequest, response: AsyncHttpServerResponse ->
+            response.headers.add("Access-Control-Allow-Origin", "*")
+            response.headers.add("Access-Control-Allow-Methods", "PUT, DELETE")
+            response.code(200)
+            response.send("")
+        }
+
     private val deleteData =
         HttpServerRequestCallback { request: AsyncHttpServerRequest, response: AsyncHttpServerResponse ->
-            val mapName: String? = request.query.getString("map")
+            val mapName: String? = (if (request.path.contains("/map/")) request.path.substringAfterLast("/")
+            else request.query.getString("map"))
```

Comment:
I don't understand why `String?` while there is always a value?

Reply:
request.query.getString("map") can be null